ambari-metrics/ambari-metrics-timelineservice/src/main/java/org/apache/ambari/metrics/core/timeline/discovery/TimelineMetricMetadataManager.java

```diff
@@ -453,17 +452,19 @@ private MetricUuidGenStrategy getUuidStrategy(Configuration configuration) {
     }
 
     byte[] uuid = uuidGenStrategy.computeUuid(hostname, HOSTNAME_UUID_LENGTH);
```

Comment:
What about forward lookup? I thought we skip uuid computation if its in the map

Reply:
Yes, we do it in lines 443-445.